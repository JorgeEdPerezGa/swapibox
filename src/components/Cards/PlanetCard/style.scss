.card-title-planet {
  padding-right: 35px;
  background: #1f413d;
}
