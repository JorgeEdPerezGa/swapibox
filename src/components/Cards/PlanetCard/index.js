import React from 'react';
import FavoriteButton from '../../FavoriteButton';
import PropTypes from 'prop-types';
import './style.css';

const PlanetCard = ({ card, name, climate, population, residents, favorite, handleFavorites }) => {
  const favortieClass = favorite ? 'favorited' : '';

  return (
    <article className = 'card'>
      <FavoriteButton
        favortieClass = { favortieClass }
        handleFavorites = { handleFavorites }
        card = { card }/>
      <h1 className = 'card-title card-title-planet'>{ name }</h1>
      <p>climate: { climate }</p>
      <p>population: { population }</p>
      <p>residents: { residents } </p>
    </article>
  );
};

PlanetCard.propTypes = {
  handleFavorites: PropTypes.func,
  card: PropTypes.object,
  name: PropTypes.string,
  climate: PropTypes.string,
  population: PropTypes.string,
  residents: PropTypes.string,
  favorite: PropTypes.string
};

export default PlanetCard;
