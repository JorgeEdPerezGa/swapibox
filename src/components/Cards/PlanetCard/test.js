import React from 'react';
import PlanetCard from './index';
import setupTest from '../../../setupTests';
import { shallow } from 'enzyme';

describe('<PlanetCard/>', () => {
  const wrapper = shallow(<PlanetCard />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
