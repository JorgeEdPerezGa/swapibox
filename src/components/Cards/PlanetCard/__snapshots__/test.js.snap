// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetCard/> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FavoriteButton
          card={undefined}
          favortieClass=""
          handleFavorites={undefined}
/>,
        <h1
          className="card-title card-title-planet"
/>,
        <p>
          climate: 
</p>,
        <p>
          population: 
</p>,
        <p>
          residents: 
           
</p>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "card": undefined,
          "favortieClass": "",
          "handleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "card-title card-title-planet",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "climate: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "climate: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "population: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "residents: ",
            undefined,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          "residents: ",
          undefined,
          " ",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FavoriteButton
            card={undefined}
            favortieClass=""
            handleFavorites={undefined}
/>,
          <h1
            className="card-title card-title-planet"
/>,
          <p>
            climate: 
</p>,
          <p>
            population: 
</p>,
          <p>
            residents: 
             
</p>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "card": undefined,
            "favortieClass": "",
            "handleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-title card-title-planet",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "climate: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "climate: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "population: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "residents: ",
              undefined,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            "residents: ",
            undefined,
            " ",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
